compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -928,9 +927,56 @@ trait Implicits:
           if normalImports.isEmpty then suggestedImports else normalImports
         end hiddenImplicitsAddendum
 
-        msg(userDefined.getOrElse(
-          em"no implicit argument of type $pt was found${location("for")}"))() ++
-        hiddenImplicitsAddendum
+        val userDefinedParamMessage = paramSymWithMethodTree.flatMap{ (sym, tree) =>
```

Comment:
Here I would add a comment with an example of an implicitNotFound on a parameter so the reader knows what we're looking for.